tests/lib/nested.sh

```diff
@@ -1272,39 +1282,22 @@ nested_start_core_vm_unit() {
         PARAM_ASSERTIONS="-drive if=none,id=stick,format=raw,file=$NESTED_ASSETS_DIR/assertions.disk,cache=none,format=raw -device nec-usb-xhci,id=xhci -device usb-storage,bus=xhci.0,removable=true,drive=stick"
     fi
     if nested_is_core_ge 20; then
-        # use a bundle EFI bios by default
-        local OVMF_CODE OVMF_VARS
-        OVMF_CODE=""
-        OVMF_VARS=""
-
-        if nested_is_core_ge 22; then
-            wget -q https://storage.googleapis.com/snapd-spread-tests/dependencies/OVMF_CODE.secboot.fd
-            mv OVMF_CODE.secboot.fd /usr/share/OVMF/OVMF_CODE.secboot.fd
-            wget -q https://storage.googleapis.com/snapd-spread-tests/dependencies/OVMF_VARS.snakeoil.fd
-            mv OVMF_VARS.snakeoil.fd /usr/share/OVMF/OVMF_VARS.snakeoil.fd
-            wget -q https://storage.googleapis.com/snapd-spread-tests/dependencies/OVMF_VARS.ms.fd
-            mv OVMF_VARS.ms.fd /usr/share/OVMF/OVMF_VARS.ms.fd
-            OVMF_CODE="_4M"
-            OVMF_VARS="_4M"
+        nested_ensure_ovmf
+        local OVMF_CODE OVMF_VARS OVMF_VARS_SECBOOT OVMF
+        if os.query is-arm; then
+            OVMF=QEMU
+        else
+            OVMF=OVMF
         fi
+        OVMF_CODE="${NESTED_ASSETS_DIR}/ovmf/fw/${OVMF}_CODE.fd"
+        OVMF_VARS="${NESTED_ASSETS_DIR}/ovmf/fw/${OVMF}_VARS.fd"
+        OVMF_VARS_SECBOOT="${NESTED_ASSETS_DIR}/ovmf/fw/${OVMF}_VARS.enrolled.fd"
 
         if nested_is_secure_boot_enabled; then
-            OVMF_CODE=".secboot"
-            if [ "$NESTED_FORCE_MS_KEYS" != "true" ] && [ "$NESTED_BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
-                OVMF_VARS=".snakeoil"
-            else
-                OVMF_VARS=".ms"
-            fi
-        fi
-
-        if os.query is-arm; then
-            cp -f "/usr/share/AAVMF/AAVMF_VARS.fd" "$NESTED_ASSETS_DIR/AAVMF_VARS.fd"
-            PARAM_BIOS="-drive file=/usr/share/AAVMF/AAVMF_CODE.fd,if=pflash,format=raw,unit=0,readonly=on -drive file=$NESTED_ASSETS_DIR/AAVMF_VARS.fd,if=pflash,format=raw"
-        else
-            cp -f "/usr/share/OVMF/OVMF_VARS${OVMF_VARS}.fd" "$NESTED_ASSETS_DIR/OVMF_VARS${OVMF_VARS}.fd"
-            PARAM_BIOS="-drive file=/usr/share/OVMF/OVMF_CODE${OVMF_CODE}.fd,if=pflash,format=raw,unit=0,readonly=on -drive file=$NESTED_ASSETS_DIR/OVMF_VARS${OVMF_VARS}.fd,if=pflash,format=raw"
-            PARAM_MACHINE="-machine q35${ATTR_KVM} -global ICH9-LPC.disable_s3=1"
+            OVMF_VARS="${OVMF_VARS_SECBOOT}"
         fi
+        PARAM_BIOS="-drive file=${OVMF_CODE},if=pflash,format=raw,readonly=on -drive file=${OVMF_VARS},if=pflash,format=raw"
```

Comment:
Before we got a new copy of the vars blob for each run, and optionally, only in the fde branch, the state file could be kept around. however, with the changes right now we'll reuse the vars blob file each time. IMO it'd be better to keep the old behavior of starting with a clean copy.
```suggestion
        OVMF_VARS_TMP="${OVMF_VARS}.tmp"
        # make a copy of the original VARS blob, so that we start with a clean slate
        cp -fv "$OVMF_VARS" "$OVMF_VARS_TMP"
        PARAM_BIOS="-drive file=${OVMF_CODE},if=pflash,format=raw,readonly=on -drive file=${OVMF_VARS_TMP},if=pflash,format=raw"
```

Reply:
I suppose I should reset unless `NESTED_KEEP_FIRMWARE_STATE` is defined.

Reply:
Now I realize we are missing stuff from the FDE branch. For example d016c599670255bd59ca6231c046ad9000677207

Reply:
this is fine, we can either cherry pick it or simply add it in the fde branch during rebase

Reply:
Cherry picked the commits needed and fixed the commit.